inject-java/src/test/groovy/io/micronaut/inject/executable/ExecutableBeanSpec.groovy

```diff
@@ -102,6 +105,118 @@ class MyBean {
+
+    @Executable
+    public void saveAll(@Valid List<io.micronaut.inject.executable.Book> books) {
+    }
+
+    @Executable
+    public <T extends io.micronaut.inject.executable.Book> void saveAll2(@Valid List<? extends T> book) {
+    }
+
+    @Executable
+    public <T extends io.micronaut.inject.executable.Book> void saveAll3(@Valid List<T> book) {
+    }
+
+    @Executable
+    public void save2(@Valid io.micronaut.inject.executable.Book book) {
+    }
+
+    @Executable
+    public <T extends io.micronaut.inject.executable.Book> void save3(@Valid T book) {
+    }
+
+    @Executable
+    public io.micronaut.inject.executable.Book get() {
+        return null;
+    }
+}
+
+''')
+        when:
+            def saveAll = definition.findMethod("saveAll", List.class).get()
+            def listTypeArgument = saveAll.getArguments()[0].getTypeParameters()[0]
+        then:
+            !saveAll.hasAnnotation(RequiresValidation)
+            !saveAll.hasStereotype(RequiresValidation)
+            listTypeArgument.getAnnotationMetadata().hasAnnotation(MyEntity.class)
+            listTypeArgument.getAnnotationMetadata().hasAnnotation(Introspected.class)
+            listTypeArgument.getAnnotationMetadata().hasStereotype(Introspected.class)
+            !listTypeArgument.getAnnotationMetadata().hasAnnotation(BeanProperties.class)
+            !listTypeArgument.getAnnotationMetadata().hasStereotype(BeanProperties.class)
+
+        when:
+            def saveAll2 = definition.findMethod("saveAll2", List.class).get()
+            def listTypeArgument2 = saveAll2.getArguments()[0].getTypeParameters()[0]
+        then:
+            !saveAll2.hasAnnotation(RequiresValidation)
+            !saveAll2.hasStereotype(RequiresValidation)
+            listTypeArgument2.getAnnotationMetadata().hasAnnotation(MyEntity.class)
+            listTypeArgument2.getAnnotationMetadata().hasAnnotation(Introspected.class)
+            listTypeArgument2.getAnnotationMetadata().hasStereotype(Introspected.class)
+            !listTypeArgument2.getAnnotationMetadata().hasAnnotation(BeanProperties.class)
+            !listTypeArgument2.getAnnotationMetadata().hasStereotype(BeanProperties.class)
+
+        when:
+            def saveAll3 = definition.findMethod("saveAll3", List.class).get()
+            def listTypeArgument3 = saveAll3.getArguments()[0].getTypeParameters()[0]
+        then:
+            !saveAll3.hasAnnotation(RequiresValidation)
+            !saveAll3.hasStereotype(RequiresValidation)
+            listTypeArgument3.getAnnotationMetadata().hasAnnotation(MyEntity.class)
+            listTypeArgument3.getAnnotationMetadata().hasAnnotation(Introspected.class)
+            listTypeArgument3.getAnnotationMetadata().hasStereotype(Introspected.class)
+            !listTypeArgument3.getAnnotationMetadata().hasAnnotation(BeanProperties.class)
+            !listTypeArgument3.getAnnotationMetadata().hasStereotype(BeanProperties.class)
+// TODO: validate this behaviour
```

Comment:
I will keep it and address it in the next PR, I don't think cases above are correct.